### ISSUE TITLE
Adding the "Has Attachment" column in the Emails module slows it down

### ISSUE THREAD

**Issue by satnav** [salesagility/SuiteCRM#7660]
#### Issue

In general, a mailbox takes about 4-5 seconds to load in the Emails module. However, once I add the column "Has Attachments" the time it takes to load the module increases to about 30 seconds.

#### Expected Behavior

Loading the emails module should take a similar amount of time as if the the "Has Attachments" column wasn't active.

#### Actual Behavior

The time it takes to load the module goes up to 30 seconds.

I also have an assortment of warning/error messages:
[WARN] Email ID is Empty
[DEPRECATED] Using row number in fetchByAssoc is not portable and no longer supported. Please fix your code.
[WARN] Trying to get property of non-object: list view data "order by" is not defined
[WARN] Trying to get property of non-object: list view data "sort order" is not defined
[WARN] Empty or undefined Email Folder ID
[WARN] unknown or undefined folder type (we will use 'imap' instead): inbound
[WARN] Request ssl value not found.
[WARN] Requested folder is not defined
[ERROR] ImapHandler trying to use a non valid resource stream.
[WARN] Undefined index: server_unique_key
[ERROR] Duplicated folder detected: <id>

#### Possible Fix

I'm not familiar with the code, I can't offer a fix...

#### Steps to Reproduce

1. Set up an Inbound Email (All my inbound emails were group inboxes)
2. Go to the emails module
3. Move "Has Attachments" from "Hidden" to "Displayed"
4. Reload the emails module

#### Context

If the module takes 30 seconds to load, and then clicking on each email to open it takes another 30 seconds to load, the module is unusable. The workaround for now is to not show the column, but then you can't see whether an email has an attachment unless you import it.

#### Your Environment

* SuiteCRM Version used:
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version (e.g. MySQL, PHP 7):
* Operating System and version (e.g Ubuntu 16.04):
SuiteCRM Version: 7.11.6
Browser: Chrome: 75.0.3770.142
Environment: MySQL 5.7.22, PHP 7.3
Operating System: Amazon Linux 2 (Cloud hosted)

**Comment by Mausino** [comment]
@satnav i agree, i also don't know why sometimes it tooks 3-4 second and why sometimes almost one minute

**Comment by Dillon-Brown** [comment]
Yes, this is expected. If you have the attachment icon visible in the list-view you will take a performance hit as it will need to pull down the entire email from IMAP (Including any attachments) for every email visible on the list-view.

The only real ways to improve this would go against the design philosophy behind the email module (Caching, etc). I can raise this as a suggestion as it might be possible to shave off a few seconds but it won't be possible to include the has_attachment field without taking any performance hits.
